cmd/zetatool/filterdeposit/evm.go

```diff
@@ -0,0 +1,208 @@
+func GetEthHashList(cfg *config.Config) []Deposit {
+	startBlock := cfg.EvmStartBlock
+	client, err := ethclient.Dial(cfg.EthRPC)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Connection successful")
+
+	header, err := client.HeaderByNumber(context.Background(), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	latestBlock := header.Number.Uint64()
+	fmt.Println("latest Block: ", latestBlock)
+
+	endBlock := startBlock + cfg.EvmMaxRange
+	deposits := make([]Deposit, 0)
+	segment := 0
+	for startBlock < latestBlock {
+		fmt.Printf("adding segment: %d, startblock: %d\n", segment, startBlock)
+		deposits = append(deposits, GetHashListSegment(client, startBlock, endBlock, cfg)...)
+		startBlock = endBlock
+		endBlock = endBlock + cfg.EvmMaxRange
+		if endBlock > latestBlock {
+			endBlock = latestBlock
+		}
+		segment++
+	}
+	return deposits
+}
+
+// GetHashListSegment queries and filters deposits for a given range
+func GetHashListSegment(client *ethclient.Client, startBlock uint64, endBlock uint64, cfg *config.Config) []Deposit {
+	deposits := make([]Deposit, 0)
+
+	connectorAddress := common.HexToAddress(cfg.ConnectorAddress)
+	connectorContract, err := zetaconnector.NewZetaConnectorNonEth(connectorAddress, client)
+	if err != nil {
+		fmt.Println("error: ", err.Error())
+	}
+	erc20CustodyAddress := common.HexToAddress(cfg.CustodyAddress)
+	erc20CustodyContract, err := erc20custody.NewERC20Custody(erc20CustodyAddress, client)
+	if err != nil {
+		fmt.Println("error: ", err.Error())
+	}
+
+	custodyIter, err := erc20CustodyContract.FilterDeposited(&bind.FilterOpts{
+		Start:   startBlock,
+		End:     &endBlock,
+		Context: context.TODO(),
+	}, []common.Address{})
+	if err != nil {
+		fmt.Println("error loading filter: ", err.Error())
+		return deposits
+	}
+
+	connectorIter, err := connectorContract.FilterZetaSent(&bind.FilterOpts{
+		Start:   startBlock,
+		End:     &endBlock,
+		Context: context.TODO(),
+	}, []common.Address{}, []*big.Int{})
+	if err != nil {
+		fmt.Println("error loading filter: ", err.Error())
+		return deposits
+	}
+
+	// ********************** Get ERC20 Custody Deposit events
+	for custodyIter.Next() {
+		// sanity check tx event
+		err := CheckEvmTxLog(&custodyIter.Event.Raw, erc20CustodyAddress, "", TopicsDeposited)
+		if err == nil {
+			//fmt.Println("adding deposits")
+			deposits = append(deposits, Deposit{
+				TxID:   custodyIter.Event.Raw.TxHash.Hex(),
+				Amount: custodyIter.Event.Amount.Uint64(),
+			})
+		}
+	}
+
+	// ********************** Get Connector ZetaSent events
+	for connectorIter.Next() {
+		// sanity check tx event
+		err := CheckEvmTxLog(&connectorIter.Event.Raw, connectorAddress, "", TopicsZetaSent)
+		if err == nil {
+			//fmt.Println("adding deposits")
+			deposits = append(deposits, Deposit{
+				TxID:   connectorIter.Event.Raw.TxHash.Hex(),
+				Amount: connectorIter.Event.ZetaValueAndGas.Uint64(),
+			})
+		}
+	}
+
+	//********************** Get Transactions sent directly to TSS address
```

Comment:
```suggestion
	// Get ERC20 Custody Deposit events
	for custodyIter.Next() {
		// sanity check tx event
		err := CheckEvmTxLog(&custodyIter.Event.Raw, erc20CustodyAddress, "", TopicsDeposited)
		if err == nil {
			//fmt.Println("adding deposits")
			deposits = append(deposits, Deposit{
				TxID:   custodyIter.Event.Raw.TxHash.Hex(),
				Amount: custodyIter.Event.Amount.Uint64(),
			})
		}
	}

	// Get Connector ZetaSent events
	for connectorIter.Next() {
		// sanity check tx event
		err := CheckEvmTxLog(&connectorIter.Event.Raw, connectorAddress, "", TopicsZetaSent)
		if err == nil {
			//fmt.Println("adding deposits")
			deposits = append(deposits, Deposit{
				TxID:   connectorIter.Event.Raw.TxHash.Hex(),
				Amount: connectorIter.Event.ZetaValueAndGas.Uint64(),
			})
		}
	}

	// Get Transactions sent directly to TSS address
```
Why these prefix in comment?
If this represent a specific separate logic in the function, we shoudl create sub functions